streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -114,6 +114,7 @@ public class RocksDBStore implements KeyValueStore<Bytes, byte[]>, BatchWritingS
     private boolean userSpecifiedStatistics = false;
 
     private final RocksDBMetricsRecorder metricsRecorder;
+    private final boolean userManagedIterators;
```

Comment:
"User" sounds a little bit like KS user to me. Maybe we can rename, eg, `selfManagedIterators` or similar (not sure what a good name would be).

Reply:
Good point. I think `selfManagedIterators` is also confusing though because it's unclear whether "self" means the store itself or the caller themselves.

I've updated this to `autoManagedIterators` which means the opposite of what I initially had (for `userManagedIterators`) and added a comment.